### PR TITLE
New package: libknet1 / libknet1-devel

### PR DESCRIPTION
**Pull request by jirib**
needed for corosync package, I'm interested only in the lib, so dropping daemon and SCTP as well (no use and cannot test).

### CODE REVIEW

File: srcpkgs/libknet1/template [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+pkgname=libknet1
+version=1.11
+revision=1
+wrksrc=kronosnet-${version}
+build_style=gnu-configure
+configure_args="--disable-crypto-nss --disable-dependency-tracking
+ --disable-libknet-sctp --disable-libnozzle --disable-kronosnetd --disable-poc
+ --disable-static"
+hostmakedepends="autoconf automake libtool pkg-config"
+makedepends="bzip2-devel doxygen liblz4-devel liblzma-devel libqb-devel
```
Comment (Hoshpak) [inline srcpkgs/libknet1/template]: doxygen should be in `hostmakedepends`.

File: srcpkgs/libknet1/template [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+	rm -rf ${DESTDIR}/usr/share/doc/kronosnet
+	vdoc README
+}
+
+libknet1-devel_package() {
+	license="LGPL-2.1-only"
+	short_desc="${short_desc} (development files)"
+	pkg_install() {
+		vmove usr/include
+		vmove "usr/lib/pkgconfig/*"
```
Comment (Hoshpak) [inline srcpkgs/libknet1/template]: Why not just vmove the whole directory? This way it leaves an empty directory which then has to be ignored for packaging.

File: srcpkgs/libknet1/template [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+ libxml2-devel libzstd-devel lzo-devel zlib-devel"
+short_desc="Kronosnet core switching implementation (libraries)"
+maintainer="Jiří Bělka <jirib79@gmail.com>"
+license="GPL-2.0-or-later"
+homepage="https://www.kronosnet.org"
+distfiles="https://kronosnet.org/releases/kronosnet-${version}.tar.bz2"
+checksum=c2ee20a9b5b0791ed5379b38b0b51d549603933925a839f045455553f03e6114
+
+post_install() {
+	vmkdir usr/share/licenses/${pkgname}
```
Comment (Hoshpak) [inline srcpkgs/libknet1/template]: Please remove this line, it will just create an unneeded empty directory now.

**Comment by jirib** [comment]
> pkglint reports
> 
> ```
> => WARNING: usr/lib/*.so should be in -devel package
> ```
> 
> Please move the unversioned symlink to the -devel subpackage. This can be done e.g. with:
> 
> ```
> vmove "usr/lib/*.so"
> ```

Done.

File: srcpkgs/libknet1/template [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+distfiles="https://kronosnet.org/releases/kronosnet-${version}.tar.bz2"
+checksum=c2ee20a9b5b0791ed5379b38b0b51d549603933925a839f045455553f03e6114
+
+post_install() {
+	rm -rf ${DESTDIR}/usr/share/doc/kronosnet
+	vdoc README
+}
+
+libknet1-devel_package() {
+	license="LGPL-2.1-only"
```
Comment (Johnnynator) [inline srcpkgs/libknet1/template]: Why is `-devel` differently licensed from the main package?

File: srcpkgs/libknet1/template [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+	short_desc="${short_desc} (development files)"
+	pkg_install() {
+		vmove usr/include
+		vmove "usr/lib/*.so"
+		vmove "usr/lib/pkgconfig"
+		vmove usr/share/man/man3
+		mkdir -p ${PKGDESTDIR}/usr/share/doc/libknet1-devel
+		cp README ${PKGDESTDIR}/usr/share/doc/libknet1-devel
```
Comment (Johnnynator) [inline srcpkgs/libknet1/template]: No need to duplicate the README, `libknet1-devel` should depend on `libknet1>=${version}_${revision}`

File: srcpkgs/libknet1/template (hunk 4 above)
Comment (Hoshpak) [reply srcpkgs/libknet1/template]: Indeed. If you are unsure if all code we are packaging is under LGPL, you could just list both licenses separated by a comma. In any case the license information should be consistent for both packages.

**Comment by jirib** [comment]
> please address the points raised by @Johnnynator .

Done - both license, -devel depends on main package and README files removed from -devel subpackage.

**Comment by Hoshpak** [comment]
Great, thanks for bearing with us. I think that's it.

**Comment by q66** [comment]
What's the `glibc-devel` dependency about? Is that necessary? If yes, and it's needed to prevent compiling on musl, why not use `archs="~*-musl"`? And if no, then it's wrong too...

**Comment by jirib** [comment]
> What's the `glibc-devel` dependency about (on host, even...)? Is that necessary? If yes, and it's needed to prevent compiling on musl, why not use `archs="~*-musl"`? And if no, then it's wrong too...

I have no skills in this area, any tip how I could proceed or self-study in this area?

**Comment by q66** [comment]
That's fine, but this template was reviewed, so unless this is somehow intentional, this should've been caught in review. You don't need a `glibc-devel` dependency, since that's a part of the base system in your `masterdir` by default. Furthermore, you definitely don't need it on host (i.e. in `hostmakedepends`) as the target (`makedepends`) is what you're compiling for; the host/target dependencies distinction is for cross-compiling, as when cross-compiling, you always need some runnable toolset (your host system won't run binaries for the target architecture you're cross-compiling to). Also, we support also `musl` based installations rather than `glibc`, and having an unneeded dependency on it prevents building on those systems. And if the program does not compile for `musl`, it can be specified using other means (for example, using `archs` like i mentioned)

**Comment by Johnnynator** [comment]
I cleaned up the template a bit and set musl to broken.
https://github.com/void-linux/void-packages/commit/d1a69567dccbb59a9214d7435c48f417b29ab99b

**Comment by q66** [comment]
Cool, but you should use the `=` operator with `[]`, not `==`

**Pull request merged**